modules/apps/oauth2-provider/oauth2-provider-shortcut/src/main/java/com/liferay/oauth2/provider/shortcut/internal/instance/lifecycle/FragmentRendererPortalInstanceLifecycleListener.java

```diff
@@ -64,7 +63,7 @@ public void portalInstanceRegistered(Company company) throws Exception {
 		User user = _userLocalService.getDefaultUser(company.getCompanyId());
 
 		oAuth2Application = _oAuth2ApplicationLocalService.addOAuth2Application(
-			company.getCompanyId(), user.getUserId(), user.getScreenName(),
```

Comment:
hey @martamedio, 
sorry this review took so long...
we should be setting the ThreadLocal in this places now, shouldn't we? I would be curious to know actually what company has the CompanyThreadLocal set in this case, if any...

Reply:
nevermind about this... it turns out it is set here: https://github.com/liferay/liferay-portal/blob/5c25dfdb42064c234dea22c3987b3253d4e1da36/modules/apps/portal/portal-instance-lifecycle-impl/src/main/java/com/liferay/portal/instance/lifecycle/internal/PortalInstanceLifecycleListenerManagerImpl.java#L138

but we don't set it when the company is removed, so we would need to send a patch to that... even though we are nout using it yet, but to make it consistent